openedx/core/djangoapps/profile_images/images.py

```diff
@@ -0,0 +1,130 @@
+        }
+    }
+
+    if uploaded_file.size > settings.PROFILE_IMAGE_MAX_BYTES:
+        raise ImageValidationError(DevMsg.FILE_TOO_LARGE)
+    elif uploaded_file.size < settings.PROFILE_IMAGE_MIN_BYTES:
+        raise ImageValidationError(DevMsg.FILE_TOO_SMALL)
+
+    # check the file extension looks acceptable
+    filename = str(uploaded_file.name).lower()
+    filetype = [ft for ft in image_types if any(filename.endswith(ext) for ext in image_types[ft]['extension'])]
+    if not filetype:
+        raise ImageValidationError(DevMsg.FILE_BAD_TYPE)
+    filetype = filetype[0]
+
+    # check mimetype matches expected file type
+    if uploaded_file.content_type not in image_types[filetype]['mimetypes']:
+        raise ImageValidationError(DevMsg.FILE_BAD_MIMETYPE)
+
+    # check image file headers match expected file type
+    headers = image_types[filetype]['magic']
+    if uploaded_file.read(len(headers[0]) / 2).encode('hex') not in headers:
+        raise ImageValidationError(DevMsg.FILE_BAD_EXT)
+    # avoid unexpected errors from subsequent modules expecting the fp to be at 0
+    uploaded_file.seek(0)
+
+
+def _get_scaled_image_file(image_obj, size):
+    """
+    Given a PIL.Image object, get a resized copy using `size` (square) and
+    return a file-like object containing the data saved as a JPEG.
+
+    Note that the file object returned is a django ContentFile which holds
+    data in memory (not on disk).
+    """
+    if image_obj.mode != "RGB":
+        image_obj = image_obj.convert("RGB")
+    scaled = image_obj.resize((size, size), Image.ANTIALIAS)
+    string_io = StringIO()
+    scaled.save(string_io, format='JPEG')
+    image_file = ContentFile(string_io.getvalue())
+    return image_file
+
+
+def generate_profile_images(image_file, profile_image_names):
+    """
+    Generates a set of image files based on image_file and
+    stores them according to the sizes and filenames specified
+    in `profile_image_names`.
+    """
+    image_obj = Image.open(image_file)
+
+    # first center-crop the image if needed (but no scaling yet).
+    width, height = image_obj.size
+    if width != height:
+        side = width if width < height else height
+        image_obj = image_obj.crop(((width - side) / 2, (height - side) / 2, (width + side) / 2, (height + side) / 2))
+
+    storage = get_profile_image_storage()
+    for size, name in profile_image_names.items():
+        scaled_image_file = _get_scaled_image_file(image_obj, size)
+        # Store the file.
+        # TODO overwrites should be atomic, but FileStorage doesn't support this.
+        if storage.exists(name):
+            storage.delete(name)
+        storage.save(name, scaled_image_file)
+
+
+def remove_profile_images(profile_image_names):
+    """
+    Physically remove the image files specified in `profile_image_names`
+    """
+    storage = get_profile_image_storage()
+    for name in profile_image_names.values():
+        storage.delete(name)
```

Comment:
Sorry for the late response. I was surprised to see that the delete method is not documented to raise anything.